File: typescript/src/examples/solana/getAccount.ts
// Usage: pnpm tsx src/examples/solana/getAccount.ts

import { config } from "dotenv";

import { CdpClient } from "../../client/cdp";

/**
 * This example shows how to get a Solana account
 */
async function main() {
  config();

  const cdp = new CdpClient();

  let address: string;
  try {
    const account = await cdp.solana.createAccount({ name: "Account1" });
    console.log("Successfully created Solana account:", account.address);
    address = account.address;
  } catch (error) {
    console.error("Error creating Solana account:", error);
    return;
  }

  let account = await cdp.solana.getAccount({
    address,
  });

  console.log("Solana account address:", account.address);

  account = await cdp.solana.getAccount({
    name: account.name,
  });

  console.log("Solana account name:", account.name);
}

main().catch(console.error);

File: examples/python/solana/get_account.py
# Usage: uv run python solana/get_account.py

import asyncio

from cdp import CdpClient


async def main():
    async with CdpClient() as cdp:
        account = await cdp.solana.create_account()
        account = await cdp.solana.get_account(address=account.address)
        print("Account Address: ", account.address)


asyncio.run(main())
